scripts/zfs-tests.sh

```diff
@@ -695,17 +695,21 @@ fi
 #
 # Run all the tests as specified.
 #
+EVILRES=$(mktemp -u -t test-runner-error)
 msg "${TEST_RUNNER} ${QUIET:+-q}" \
     "-c \"${RUNFILES}\"" \
     "-T \"${TAGS}\"" \
     "-i \"${STF_SUITE}\"" \
     "-I \"${ITERATIONS}\""
-${TEST_RUNNER} ${QUIET:+-q} \
+(${TEST_RUNNER} ${QUIET:+-q} \
     -c "${RUNFILES}" \
     -T "${TAGS}" \
     -i "${STF_SUITE}" \
     -I "${ITERATIONS}" \
-    2>&1 | tee "$RESULTS_FILE"
+    2>&1 ; echo $? > "$EVILRES"; )| tee "$RESULTS_FILE"
+
+RUNRESULT=$(cat "$EVILRES")
```

Comment:
`read -r RUNRESULT < "$EVILRES"` saves you a subshell and you also want to `rm "$EVILRES"`